src/main/java/org/gitlab4j/api/webhook/WebHookManager.java

```diff
@@ -98,22 +97,13 @@ public void handleEvent(HttpServletRequest request) throws GitLabApiException {
 
             fireEvent(event);
 
-        } catch (JsonParseException jpe) {
-            errorMessage = jpe.getMessage();
-            LOG.warning("Error parsing JSON data, error=" + errorMessage);
-        } catch (JsonMappingException jme) {
-            errorMessage = jme.getMessage();
-            LOG.warning("Error mapping JSON data, error=" + errorMessage);
-        } catch (IOException ioe) {
-            errorMessage = ioe.getMessage();
-            LOG.warning("Error reading JSON data, error=" + errorMessage);
+        } catch (IOException e) {
+            LOG.warning("Error parsing JSON data, error=" + e.getMessage());
+            throw new GitLabApiException(e);
         } catch (Exception e) {
-            errorMessage = e.getMessage();
-            LOG.warning("Unexpected error reading JSON data, error=" + errorMessage);
+            LOG.warning("Unexpected error reading JSON data, error=" + e.getMessage());

```

Comment:
Since you have already simplified it by removing other catch blocks, can you make the logged message more generic and only capture Exception and remove the catch (IOException e) block?  Something like this:

``` LOG.warning("Unexpected error processing JSON data, error=" + e.getMessage());```

Reply:
There is still a distinction between `IOException` which is the base of both `JsonParseException` and `JsonMappingException` and just a plain `Exception`.  So going through logs and seeing the first message tells me there was likely a parsing error, but seeing the latter message means something really unexpected happened, which i find useful.

Reply:
@tiaanl 
SInce your change sets the cause, can you not get that info from the exception returned by using getCause() ?

Reply:
How about the following, simplifies, but still gives the exact reason for the exception?
```
} catch (Exception e) {
    LOG.warning("Error processing JSON data, exception=" + e.getClass().getSimpleName() + ", error=" + e.getMessage());
    throw new GitLabApiException(e);
}
```

Reply:
Yeah that'll work perfectly.  Can see info in logs and expect error programmatically.  Added commit.